tests: samples: Moved sample.yaml from static lib to hello_world

vikramx2112 opened the pull request:
Automated test suit is not able to compile/run static lib sample app,
because Makefile.inc is not included in top level of staticlib,
due to that Make target(INITCONFIG) is not known to test suite.
To run static lib on automated test suit required to move sample.yaml
from static_lib top directory to hello_world directory.

Signed-off-by: Vikram Singh Shekhawat <vikramx.shekhawat@intel.com>

nashif reviewed (changes requested):
We are not testing hello world here, this is a test for the static_lib, so moving the sample.yaml will not work, The problem is probably in how you run samples in your environment, you probably making assumptions that are invalid?

vikramx2112 committed:
tests: samples: Moved sample.yaml from static lib to hello_world

Automated test suit is not able to compile static lib sample app,
beacause Makefile.inc is not included in top level of staticlib,
due to that Make target(INITCONFIG) is not known to test suite.
To run static lib on automated test suit required to move sample.yaml
from static_lib top directory to hello_world directory.

Signed-off-by: Vikram Singh Shekhawat <vikramx.shekhawat@intel.com>

nashif commented:
closing, this sample will be obsolete and this change is not going to work.
